### PR TITLE
[dynamo] Add is_exporting flag

### PR DESCRIPTION
**Pull request by angelayi**
* `is_exporting` returns true if we're exporting with dynamo
* `is_compiling` returns true if we're compiling or exporting with dynamo
* both flags return false in eager

cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @desertfire

### CODE REVIEW

File: test/dynamo/test_export.py [hunk 1]
```diff
@@ -2069,6 +2069,31 @@ def forward(self, *args):
         res = gm(input_tensor, input_tensor2)
         self.assertTrue(torch._dynamo.utils.same(ref, res))
 
+    @config.patch(dynamic_shapes=True)
```
Comment (tugsbayasgalan) [inline test/dynamo/test_export.py:2072]: Why does this test need dynamic_shapes=True?

File: torch/_dynamo/external_utils.py [hunk 2]
```diff
@@ -3,3 +3,7 @@
 
 def is_compiling():
     return False
+
+
+def is_exporting():
```
Comment (tugsbayasgalan) [inline torch/_dynamo/external_utils.py:8]: NIT: can you add a docstring here explaining various use cases, like how user can skip over certain part of the code etc?

**Comment by yanboliang** [comment]
@angelayi Can you share the motivation of adding this flag? Does it mean we allow FX graph divergence between ```dynamo.optimize``` and ```dynamo.export```? Do we expect this API will be used explicitly like what in your unit test?

**Comment by angelayi** [comment]
> Can you share the motivation of adding this flag? Does it mean we allow FX graph divergence between dynamo.optimize and dynamo.export? Do we expect this API will be used explicitly like what in your unit test?

The context is that currently we modified part of a model to use the control flow ops, and have that control flow code wrapped with a "if is_compling:..." so that torchscript and normal training doesn't run that control flow op. However, a user was trying to run that code with dynamo training, and since it's compiling with dynamo, it will run the cond op. However, the cond op doesn't support training so I want to add the flag "is_exporting" similar to how "is_compiling" exists for these cases.

**Pull request closed without merging**